contracts/JBTieredNFTRewardDelegate.sol

```diff
@@ -345,7 +344,7 @@ contract JBTieredLimitedNFTRewardDataSource is JBNFTRewardDataSource, IJBTieredN
     @return The ID of the token.
     @return The tier number.
   */
-  function _generateTokenId(uint256 _amount) internal returns (uint256, uint256) {
+  function _generateTokenId(uint256 _amount) internal view returns (uint256, uint256) {
```

Comment:
this shouldn't be `view` because of the `--_tier.remainingAllowance`.

I think instead of `--_tier.remainingAllowance` it should be `--_tiers[_i].remainingAllowance` to actual write to the saved property